### PR TITLE
add I2V sft and fix an error

### PR DESCRIPTION
**Pull request by jiashenggu**
i am not sure how to set ofs in training, I just follow the inference pipeline setting

### CODE REVIEW

**Comment by sayakpaul** [comment]
Thanks! Could you provide some example results as well?

File: training/cogvideox_image_to_video_lora.py [hunk 1]
```diff
@@ -799,12 +799,15 @@ def load_model_hook(models, input_dir):
                 # (this is the forward diffusion process)
                 noisy_video_latents = scheduler.add_noise(video_latents, noise, timesteps)
                 noisy_model_input = torch.cat([noisy_video_latents, image_latents], dim=2)
-
+                model_config.patch_size_t if hasattr(model_config, "patch_size_t") else None,
```
Comment (sayakpaul) [inline training/cogvideox_image_to_video_lora.py:802]: This seems wrong. No assignment to a variable. Is this expected?

File: training/cogvideox_text_to_video_lora.py [hunk 2]
```diff
@@ -778,7 +778,7 @@ def load_model_hook(models, input_dir):
             last_lr = lr_scheduler.get_last_lr()[0] if lr_scheduler is not None else args.learning_rate
             logs = {"loss": loss.detach().item(), "lr": last_lr}
             # gradnorm + deepspeed: https://github.com/microsoft/DeepSpeed/issues/4555
-            if accelerator.distributed_type != DistributedType.DEEPSPEED:
+            if accelerator.sync_gradients and accelerator.distributed_type != DistributedType.DEEPSPEED:
```
Comment (sayakpaul) [inline training/cogvideox_text_to_video_lora.py:781]: Should the OFS related changes be propagated here too?

File: training/cogvideox_text_to_video_sft.py [hunk 3]
```diff
@@ -744,7 +744,7 @@ def load_model_hook(models, input_dir):
             last_lr = lr_scheduler.get_last_lr()[0] if lr_scheduler is not None else args.learning_rate
```
Comment (sayakpaul) [inline training/cogvideox_text_to_video_sft.py:744]: Should the OFS related changes be propagated here too?

**Review by sayakpaul — COMMENTED** [review]
I left some preliminary comments. Will let @a-r-r-o-w comment on the new script first

File: train_image_to_video_sft.sh [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+OMP_NUM_THREADS=16
+
+GPU_IDS="0,1"
+
+# Training Configurations
+# Experiment with as many hyperparameters as you want!
+LEARNING_RATES=("1e-4")
+LR_SCHEDULES=("cosine_with_restarts")
+OPTIMIZERS=("adamw")
+MAX_TRAIN_STEPS=("20000")
```
[the remark this answers is not in the document]
Comment (jiashenggu) [reply train_image_to_video_sft.sh:16]: I just follow the setting of train_text_to_video_sft.sh

File: training/cogvideox_image_to_video_lora.py (hunk 1 above)
Comment (jiashenggu) [reply training/cogvideox_image_to_video_lora.py]: Forget to delete early edit. Fixed it

File: training/cogvideox_text_to_video_lora.py (hunk 2 above)
Comment (jiashenggu) [reply training/cogvideox_text_to_video_lora.py:781]: OFS only exists in image to video

File: training/cogvideox_text_to_video_sft.py (hunk 3 above)
Comment (jiashenggu) [reply training/cogvideox_text_to_video_sft.py:744]: OFS only exists in image to video

File: train_image_to_video_sft.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (sayakpaul) [reply train_image_to_video_sft.sh:16]: Have you performed any experiments yourself?

**Review by a-r-r-o-w — COMMENTED** [review]
Thank you for the script and OFS fixes! Do you have any publically available training runs to verify correctness? For example, we did a 20000 step run on T2V SFT before sharing the script, so something similar to verify I2V would be nice (even if lower number of steps)

**Comment by jiashenggu** [comment]
> Thank you for the script and OFS fixes! Do you have any publically available training runs to verify correctness? For example, we did a 20000 step run on T2V SFT before sharing the script, so something similar to verify I2V would be nice (even if lower number of steps)
I did a 24000 step run. 
valid prompt: A black-and-white animated scene featuring three characters in a static setting. Mickey Mouse-like character stands on one leg, hands on hips, with a playful expression. Center character has an exaggerated open mouth, caught in mid-motion, suggesting singing or surprise. Female character in a tutu and flower-adorned hat dances, arms raised. Background features a plain wall with scattered musical notes. The characters maintain their positions and expressions, with no changes in lighting, environment, or camera perspective, focusing on their interaction within this continuous moment.
valid image:
valid

https://github.com/user-attachments/assets/9e9d8a1d-2c48-4d29-8460-943445678295


https://github.com/user-attachments/assets/8469b293-3d91-4350-af10-b082761d9af7

**Comment by jiashenggu** [comment]
> Thank you for the script and OFS fixes! Do you have any publically available training runs to verify correctness? For example, we did a 20000 step run on T2V SFT before sharing the script, so something similar to verify I2V would be nice (even if lower number of steps)

I did a 24000 step run, but I'm not sure it met expectations, but it seemed somehow better.

valid prompt:
A black-and-white animated scene featuring three characters in a static setting. Mickey Mouse-like character stands on one leg, hands on hips, with a playful expression. Center character has an exaggerated open mouth, caught in mid-motion, suggesting singing or surprise. Female character in a tutu and flower-adorned hat dances, arms raised. Background features a plain wall with scattered musical notes. The characters maintain their positions and expressions, with no changes in lighting, environment, or camera perspective, focusing on their interaction within this continuous moment.

valid image:
valid
base model output：

https://github.com/user-attachments/assets/68c11c81-cba8-4dda-affc-97f6b6fc0420

24000 step run model output：

https://github.com/user-attachments/assets/928c627a-5d68-4448-9ba9-6ac5071194c9

**Comment by sayakpaul** [comment]
Seems definitely better to me. At least it learned semantics and better motion (IMO).

**Comment by a-r-r-o-w** [comment]
@jiashenggu Looks good to merge! Could you rebase against the main branch? Because it looks like some changes already in main our made here as well

**Review by a-r-r-o-w — APPROVED** [review]

**Comment by a-r-r-o-w** [comment]
Thank you so much for this! I've verified that it works. Please feel free to open PRs for speedups or other suggestions for improvements :)

**Pull request merged**